client/src/pages/SharesPage.js

```diff
@@ -68,17 +88,21 @@ class SharesPage extends Component {
 
     const filteredData = this.filterData(data)
 
-    const balance = _.sum(filteredData.map((v) => v.amount))
+    const balances = this.calculateBalances(filteredData)
 
     const oldestDate = minDate(data.map((item) => item.issue_date))
 
     return (
       <Layout header={(
         <Fragment>
-          <div className={classes.balance}>
-            <span className={classes.title}>Balance:</span>
-            <span className={classes.number}>{balance}</span> shares
-          </div>
+          {
+            Object.entries(balances).map(([title, balance]) => (
+              <div className={classes.balance} key={title}>
+                <span className={classes.title}>{title}:</span>
+                <span className={classes.number}>{balance}</span> shares
+              </div>
+            ))
+          }
```

Comment:
```suggestion
          <dl style={{display: 'flex', flexWrap: 'wrap'}}>
            {Object.entries(balances).map(([key, balance]) => (
              <div className={classes.balance} key={key} style={{display: 'flex'}}>
                <dt className={classes.title}>{_.capitalize(key)}:</dt>
                <dd><span className={classes.number}>{balance}</span> shares</dd>
              </div>
            ))}
          </dl>
```

This way it's less spread through the UI and connected as one information.
The `dl` is semantic element for having some terms with values for each of them.

Reply:
Yes, it makes much more sense this way.